app/views/api/layouts.py

```diff
@@ -0,0 +1,190 @@
+from flask.views import MethodView
+from flask.globals import current_app
+from marshmallow import fields
+from marshmallow_sqlalchemy.schema import SQLAlchemySchema, auto_field
+import marshmallow as ma
+
+from app.extensions.api import Blueprint
+from app.models import Layout
+
+from . import CommonSchema
+
+
+blp = Blueprint(Layout.__tablename__ + 's', __name__)
+
+
+TITLE_DESC = (
+    'Title of the layout',
+    'Filter for a layout title'
+)
+SUBTITLE_DESC = (
+    'Subtitle of the layout',
+    'Filter for a layout subtitle'
+)
+HTML_DESC = (
+    'HTML used in the layout',
+)
+CSS_DESC = (
+    'Stylesheet used in the layout',
+)
+SCRIPTS_DESC = (
+    'Scripts to be injected in the layout. Every field is either a string or a list of strings',
+)
+SCRIPT_DESC = (
+    'Script injected in the layout',
+)
+SHOW_USERS_DESC = (
+    'Show a user list in the layout',
+    'Filter for a user list being shown'
+)
+SHOW_LATENCY_DESC = (
+    'Show the current latency in the layout',
+    'Filter for latency being shown'
+)
+READ_ONLY_DESC = (
+    'Make the current room read-only',
+    'Filter for the layout being read-only'
+)
+
+EXAMPLE = dict(
+    title='Test Room',
+    subtitle='Room for testing purposes',
+    scripts={
+        'incoming-text': 'display-text',
+        'incoming-image': 'display-image',
+        'submit-message': 'send-message',
+        'print-history': 'plain-history',
+    }
+)
+
+
+# Only two schemas are needed but four are used to prettify OpenAPI Documentation
```

Comment:
I see 5 Schema classes here. Which ones are needed and which ones are for prettification?

Reply:
`LayoutSchema` is not used by marshmallow. Other schemas only inherit from this to simplify implementations. I'll add some more comments anyway.